pkg/subnet/kube/kube.go

```diff
@@ -138,10 +139,23 @@ func NewSubnetManager(ctx context.Context, apiUrl, kubeconfig, prefix, netConfPa
 
 		log.Infof("Waiting %s for node controller to sync", nodeControllerSyncTimeout)
 		err = wait.PollUntilContextTimeout(ctx, time.Second, nodeControllerSyncTimeout, true, func(context.Context) (bool, error) {
-			return sm.nodeController.HasSynced(), nil
+			ch := make(chan bool, 1)
```

Comment:
Does this need to be a buffered chan? You send from a goroutine right away and recv in the main thread so both ends have an actor, no need to buffer?

Reply:
added comments to explain the reason for using a buffered channel send